napari/_tests/test_adding_removing.py

```diff
@@ -120,7 +121,7 @@ def my_custom_callback():
     for em in layer.events.emitters.values():
         # warningEmitters are not connected when connecting to the emitterGroup
         if not isinstance(em, WarningEmitter):
-            assert len(em.callbacks) == 1
+            assert len(em.callbacks) == count_warning_events(em.callbacks) + 1
```

Comment:
@JoOkuma can you explain what is happening here? We (@DragaDoncila and I) are very confused, both before and after the change. 😂

Reply:
The constant `1` comes from the `my_custom_callback` a few lines above.

And because we're deprecating the `_edges` events, we add some additional warning events ([here](https://github.com/JoOkuma/napari/blob/ff6ac5c736372714bb6cee78344a7ca348a1d0d1/napari/layers/points/points.py#L2218)) which can be emitted when emitting their respective `_border` event, so there're additional warning callbacks set up, and this increases the `len(em.callbacks)`.

The warning events are only triggered when there's a callback on the deprecated event, so they're not emitted by default.  Note that the `_border` (new) is connected to `_edge` (old), not the way around.

Is that clear? I can rephrase it.

@Czaki helped me improve how these deprecation warnings were emitted.